### PR TITLE
feat(notifications): Add additional fields for Github and get config from halyard

### PR DESCRIPTION
**Pull request by loispostula**
Support: https://github.com/spinnaker/echo/pull/607

Should not be merged until Halyard is released with this change: https://github.com/spinnaker/halyard/pull/1374

### CODE REVIEW

**Comment by maggieneterval** [comment]
Hey @loispostula, thanks for the PR! We have an open PR to convert some of the notification components from Angular to React (https://github.com/spinnaker/deck/pull/7075), so we may have to make some adjustments to this PR after that one gets merged (sorry in advance!). Since we'd like to wait until the Halyard change is merged and released anyway, I am going to add a "do not merge" tag on this PR for now!

**Comment by loispostula** [comment]
@maggieneterval No problem, the added fields are not required for (https://github.com/spinnaker/echo/pull/607) Once (#7075) is merged, I'll adapt

**Comment by maggieneterval** [comment]
Hey @loispostula, just wanted to let you know that the PR I linked above that converted some of the components you had edited from Angular to React has been merged, so feel free to refactor whenever you're ready!

**Comment by loispostula** [comment]
@maggieneterval I've migrated it to React, Cheers!

File: app/scripts/modules/core/src/notification/selector/types/githubstatus/GithubNotificationType.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
```
Comment (maggieneterval) [inline app/scripts/modules/core/src/notification/selector/types/githubstatus/GithubNotificationType.tsx]: We typically add a newline after third-party imports per this (hard to find) conventions doc: https://github.com/spinnaker/deck/wiki/Conventions#imports
```suggestion
import * as React from 'react';

```

File: app/scripts/modules/core/src/notification/selector/types/githubstatus/GithubNotificationType.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { FormikFormField, TextInput, Validators } from 'core/presentation';
+import { INotificationTypeCustomConfig } from 'core/domain';
+
+export class GithubNotificationType extends React.Component<INotificationTypeCustomConfig> {
+  public render() {
+    const { fieldName } = this.props;
+    return (
+      <>
+        <div className="sp-margin-m-bottom">
+          <div className={'form-group'}>
```
Comment (maggieneterval) [inline app/scripts/modules/core/src/notification/selector/types/githubstatus/GithubNotificationType.tsx]: Another convention we try to follow is not using braces to wrap string literals in JSX. Think there are a few other places in this file we can make this change as well.
```suggestion
          <div className="form-group">
```

File: app/scripts/modules/core/src/notification/selector/types/githubstatus/GithubNotificationType.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            <label className={'col-md-4 sm-label-right'}>Github Repository</label>
```
Comment (maggieneterval) [inline app/scripts/modules/core/src/notification/selector/types/githubstatus/GithubNotificationType.tsx]: I *think* the label can be included as a prop of `FormikFormField` but going to defer to @Jammy-Louie for the convention here since he very kindly wrote many similar components in React.

Comment (Jammy-Louie) [reply app/scripts/modules/core/src/notification/selector/types/githubstatus/GithubNotificationType.tsx]: You are right you can include it as a `label` prop and it is recommended to do so. However it causes misalignment in the `manualExecution` modal when it is rendered.

Going to have to go back and realign everything in notifications using the `layout` prop (to do item) or realign everything in the `manualExecution` modal.

TLDR; This is fine for now..

File: app/scripts/modules/core/src/notification/selector/types/githubstatus/GithubNotificationType.tsx (hunk 1 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/notification/selector/types/githubstatus/GithubNotificationType.tsx]: i've updated the PR and added the new line.

File: app/scripts/modules/core/src/notification/selector/types/githubstatus/GithubNotificationType.tsx (hunk 3 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/notification/selector/types/githubstatus/GithubNotificationType.tsx]: updated the PR to have everything align properly with `label` as a props of `FormikFormField`

**Comment by Jammy-Louie** [comment]
looks like https://github.com/spinnaker/halyard/pull/1374 has been merged and available
